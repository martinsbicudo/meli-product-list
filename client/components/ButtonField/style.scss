.ButtonField {
  flex: 1;
  display: flex;
  border-radius: .2rem;

  @include query-max("mobile") { flex-wrap: wrap; }

  &FormField {
    @include query-max("mobile") {
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
    }

    @include query-min("mobile") {
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
    }
  }

  &Button {
    @include query-max("mobile") {
      min-width: 100%;
      border-top-left-radius: 0;
      border-top-right-radius: 0;
    }

    @include query-min("mobile") {
      flex: 0;
      border-top-left-radius: 0;
      border-bottom-left-radius: 0;
    }
  }
}
