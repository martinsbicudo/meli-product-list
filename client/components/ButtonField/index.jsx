import React from 'react';
import {
  string,
  object,
  node,
  oneOfType,
} from 'prop-types';
import classnames from 'classnames';
import FormField from '@Components/FormField';
import Button from '@Components/Button';
import style from './style.scss';

const ButtonField = ({
  className,
  children,
  formField,
  button,
}) => (
  <div
    className={classnames(
      style.ButtonField,
      className,
    )}
  >
    <FormField
      {...formField}
      className={{
        wrapper: classnames(
          style.ButtonFieldFormField,
          (formField.className && formField.className.wrapper) || '',
        ),
        input: (formField.className && formField.className.input) || '',
      }}
    />
    <Button
      {...button}
      className={classnames(
        style.ButtonFieldButton,
        button.className || '',
      )}
    >
      {children}
    </Button>
  </div>
);

ButtonField.propTypes = {
  className: string,
  children: oneOfType([string, node]).isRequired,
  formField: object,
  button: object,
};

ButtonField.defaultProps = {
  className: '',
  formField: {
    className: {
      wrapper: '',
      input: '',
    },
  },
  button: {
    className: '',
  },
};

export default ButtonField;
