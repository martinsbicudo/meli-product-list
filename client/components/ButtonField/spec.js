import ButtonField from '.';

describe('Components - <ButtonField />', () => {
  const testMessagePlaceholder = 'TEST_PLACEHOLDER';
  const testMessageChildren = 'TEST_CHILDREN';
  const spyOnClick = jest.fn();
  const initialState = {
    button: {
      onClick: spyOnClick,
      placeholder: testMessagePlaceholder,
    },
  };
  const createWrapper = (
    props = {},
    children = null,
  ) => render(
    <ButtonField
      {...{
        ...initialState,
        ...props,
      }}
    >
      {children}
    </ButtonField>,
  );

  test('Should calls onClick', () => {
    const { getByPlaceholderText } = createWrapper();

    fireEvent.click(getByPlaceholderText(testMessagePlaceholder));

    expect(spyOnClick).toHaveBeenCalledTimes(1);
    spyOnClick.mockRestore();
  });

  test('Should finds Children', () => {
    const { getByText } = createWrapper(null, <span>{testMessageChildren}</span>);

    expect(getByText(testMessageChildren)).toBeInTheDocument();
    spyOnClick.mockRestore();
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
