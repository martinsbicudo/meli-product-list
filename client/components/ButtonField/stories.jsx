import React from 'react';
import { storiesOf } from '@storybook/react';
import ButtonField from '.';
import { placeholder } from '@babel/types';

storiesOf('ButtonField', module)
  .add('With Children', () => (
    <ButtonField>
      Text
    </ButtonField>
  ))
  .add('With FormField [config]', () => (
    <ButtonField
      formField={{
        onFocus: () => console.log('onFocus was called') /* eslint-disable-line */,
        placeholder: 'Placeholder ButtonField',
      }}
    >
      Text
    </ButtonField>
  ))
  .add('With Button [config]', () => (
    <ButtonField
      button={{
        onClick: () => console.log('onClick was called') /* eslint-disable-line */,
      }}
    >
      Text
    </ButtonField>
  ));
