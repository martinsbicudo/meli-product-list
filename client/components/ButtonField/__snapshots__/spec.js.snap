// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - <ButtonField /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="ButtonField"
  >
    <div
      class="FormField ButtonFieldFormField"
    >
      <input
        class="FormFieldInput"
        type="text"
      />
    </div>
    <button
      class="Button ButtonFieldButton"
      placeholder="TEST_PLACEHOLDER"
      type="button"
    />
  </div>
</DocumentFragment>
`;
